src/uct/ib/ud/base/ud_ep.c

```diff
@@ -585,24 +698,25 @@ ucs_status_t uct_ud_ep_create_connected_common(const uct_ep_params_t *ep_params,
     ep = uct_ud_iface_cep_get_ep(iface, ib_addr, if_addr, path_index,
                                  conn_sn, 1);
     if (ep != NULL) {
-        uct_ud_ep_set_state(ep, UCT_UD_EP_FLAG_CREQ_NOTSENT);
-        ep->flags &= ~UCT_UD_EP_FLAG_PRIVATE;
-        status     = uct_ud_iface_cep_insert_ep(iface, ib_addr, if_addr,
-                                                path_index, conn_sn, ep);
+        uct_ud_ep_set_state(ep, UCT_UD_EP_FLAG_CREQ_NOTSENT |
+                                UCT_UD_EP_FLAG_TX);
+        status = uct_ud_iface_cep_insert_ep(iface, ib_addr, if_addr,
+                                            path_index, conn_sn, ep);
         if (status != UCS_OK) {
             goto err_ep_disconnect;
         }
-        goto out_set_ep;
+        goto out_send_creq_and_set_ep;
```

Comment:
peer's endpoint should have RX capability, sending CREQ ensures it

Reply:
so now both sides send creq and crep?

Reply:
CREP could be sent even for internal ep (i.e. user didn't call `uct_ep_create`).
so, anyway we will need to send CREQ later.
and sending CREQ handles the following case:
1. A creates UCT EP, it sends CREQ.
2. B creates internal UCT EP, it sends CREP.
3. A sends some data.
4. A closes UCT EP and B creates UCT EP from internal one and sends CREQ.
5. A receives CREQ, can't find EP in CEP, checks ptr_array - will find UCT EP, stops disconenctign for it.

But CREP in this case will be just dropped, since DISCONNECTED flag set.